src/App.tsx

```diff
@@ -1,56 +1,95 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/PeopleList/PeopleList';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const delay = 1000;
+  const allPeople: Person[] = peopleFromServer;
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [showList, setShowList] = useState<boolean>(false);
+
+  const filteredPeople = useMemo(() => {
+    return allPeople.filter(
+      person => person.name.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  }, [query, allPeople]);
+
+  const applySearch = useCallback(
+    debounce(value => {
+      setQuery(value);
+    }, delay),
+    [delay],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setShowList(true);
+    applySearch(event.target.value);
```

Comment:
When you instantly change the query, you have an array of people filtered without delay. I think you can create another query that will be set with a delay in debounce and filter the array. Or just delete 32 line